Tweaks for newer Stylelint 14 rules

colinrotherham opened the pull request:
Hello 👋

Hope these updates are handy

In this PR I've mopped up a few newer Stylelint v14 rule updates for:

1. Configure `stylelint@14.7.0` [`selector-not-notation`](https://stylelint.io/user-guide/rules/selector-not-notation)
2. Configure `stylelint@14.10.0` [`annotation-no-unknown`](https://stylelint.io/user-guide/rules/annotation-no-unknown)

I've also updated to the latest v14 compatible versions of:

```console
stylelint-config-standard@29
stylelint-config-standard-scss@6.1.0
```

To avoid warnings and to seamlessly run `npm link` against `govuk-frontend` I've had to bump **package-lock.json** to [`"lockfileVersion": 2`](https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json#lockfileversion) but this maintains compatibility with npm v5 and v6

colinrotherham commented:
For context, I've been working on the `govuk-frontend` Dart Sass update:

* https://github.com/alphagov/govuk-frontend/pull/3164

colinrotherham commented on package.json:
```diff
@@ -49,15 +49,15 @@
   },
   "peerDependencies": {
-    "stylelint": "^14.4.0"
+    "stylelint": "^14.14.0 || ^15.0.0"
```
The good news?

Aside from deprecation warnings, it looks like this repo is already Stylelint v15 compatible. Although removing the deprecation warnings + adopting Prettier (from [Migrating to 15.0.0](https://stylelint.io/migration-guide/to-15/)) is out of scope for this PR

I've followed the approach of `stylelint-order` by supporting both v14 and v15:

* https://github.com/hudochenkov/stylelint-order/releases/tag/6.0.2

colinrotherham replied:
I've dropped the `peerDependency` range so we can look at Stylelint v15 (without warnings) in another PR

kevindew commented:
Hey, thanks for this @colinrotherham. Sorry for the lengthy delay, I've just returned from a long period of leave and I'm somewhat the only core maintainer. I've got a lot to catch up on so may be a week or two until I get to this, but thanks so much and I'll get back to you soon.

kevindew commented:
Thanks for this @colinrotherham. I've found some time to take a look at this, most of the changes look great. I am however a little reluctant to do a release that handles both Stylelint 14 and 15, I feel it'd be better to do one just for Stylelint 15 (I've started pulling that together in: https://github.com/alphagov/stylelint-config-gds/pull/32 with commits from this PR).

The big thing that seems a pain is the removal of stylistic rules for the adoption of prettier, which seems to be a case of working out if GDS are happier with prettier and how we provide migration guidance.

P.S Are you working at GDS? We should, if you don't already have it, get your write access to this repo.

colinrotherham commented:
@kevindew Yeah send me a message on Slack, I'm happy to help and/or split this up to assist 😊

colinrotherham commented:
Hi @kevindew I've added a CHANGELOG entry and dropped the Stylelint v15 `peerDependency`

Happy to make further changes, tweak wording etc

Thanks again

kevindew reviewed (approved):
Great - this looks good to me
